From 5f66f95914025483546d053be55f4730fd615ece Mon Sep 17 00:00:00 2001
From: "Eugene San (eugenesan)" <eugenesan@gmail.com>
Date: Sun, 23 Dec 2012 18:26:01 +0200
Subject: [PATCH] optional encryption password sync

Option to use 2 different tokens for encryption and locking.
Allows convenient usage and strong encryption at the same time.

Inspired by reading next:
https://santoku-linux.com/howto/mobile-forensics/how-to-brute-force-android-encryption

Related discussion:
http://code.google.com/p/android/issues/detail?id=29468

Change-Id: I71a31e7a04b84910d5be317e6f531c41d7ad002f

Conflicts:
	src/com/android/settings/SecuritySettings.java
---
 res/values/strings.xml                         | 2 ++
 res/xml/security_settings_encrypted.xml        | 5 +++++
 src/com/android/settings/SecuritySettings.java | 3 +++
 3 files changed, 10 insertions(+)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index 3f8e41f..396c6be 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -2449,6 +2449,8 @@
     <string name="lockpattern_too_many_failed_confirmation_attempts_footer">Try again in <xliff:g id="number">%d</xliff:g> seconds.</string>
     <!-- Displayed when user launches a widget configuration activity that was uninstalled -->
     <string name="activity_not_found">Application is not installed on your phone.</string>
+    <!-- Security & location settings screen, setting check box title. This controls whether to sync encryption password with lock screen token [CHAR LIMIT=28] -->
+    <string name="lockpattern_settings_enable_sync_enc_pass">Sync encryption password</string>
 
     <!-- ChooseLockPatternTutorial --> <skip />
     <!-- ChooseLockPatternTutorial, button labels: This is to cancel the tutorial -->
diff --git a/res/xml/security_settings_encrypted.xml b/res/xml/security_settings_encrypted.xml
index e82dc56..8d2696e 100644
--- a/res/xml/security_settings_encrypted.xml
+++ b/res/xml/security_settings_encrypted.xml
@@ -26,6 +26,11 @@
             android:summary="@string/crypt_keeper_encrypted_summary"
         />
 
+        <CheckBoxPreference
+            android:key="lock_sync_encryption_password"
+            android:title="@string/lockpattern_settings_enable_sync_enc_pass"
+        />
+
     </PreferenceCategory>
 
 
diff --git a/src/com/android/settings/SecuritySettings.java b/src/com/android/settings/SecuritySettings.java
index 59cd110..4dbe04c 100644
--- a/src/com/android/settings/SecuritySettings.java
+++ b/src/com/android/settings/SecuritySettings.java
@@ -79,6 +79,7 @@ public class SecuritySettings extends SettingsPreferenceFragment
     private static final String KEY_POWER_INSTANTLY_LOCKS = "power_button_instantly_locks";
     private static final String KEY_CREDENTIALS_MANAGER = "credentials_management";
     private static final String PACKAGE_MIME_TYPE = "application/vnd.android.package-archive";
+    private static final String LOCK_SYNC_ENCRYPTION_PASSWORD = "lock_sync_encryption_password";
 
     DevicePolicyManager mDPM;
 
@@ -458,6 +459,8 @@ public class SecuritySettings extends SettingsPreferenceFragment
             lockPatternUtils.setVisiblePatternEnabled(isToggled(preference));
         } else if (KEY_POWER_INSTANTLY_LOCKS.equals(key)) {
             lockPatternUtils.setPowerButtonInstantlyLocks(isToggled(preference));
+        } else if (LOCK_SYNC_ENCRYPTION_PASSWORD.equals(key)) {
+            lockPatternUtils.setSyncEncryptionPassword(isToggled(preference));
         } else if (preference == mShowPassword) {
             Settings.System.putInt(getContentResolver(), Settings.System.TEXT_SHOW_PASSWORD,
                     mShowPassword.isChecked() ? 1 : 0);
-- 
1.8.0.3

